python/dask_cudf/dask_cudf/groupby.py

```diff
@@ -64,51 +64,183 @@ def __getitem__(self, key):
         g._meta = g._meta[key]
         return g
 
+    @_dask_cudf_nvtx_annotate
+    def count(self, split_every=None, split_out=1):
+        if _groupby_supported(self):
```

Comment:
[Mostly a note to myself, but feel free to confirm]: `_groupby_supported` is used here to protect against the possibility that the object being grouped may not be a Dask-DataFrame collection, and that the `by` may not be a aggregation name (or list of aggregation names).

Rather than adding `if _groupby_supported(self): ... return super().<method>(split_every=split_every, split_out=split_out)` all over this file, perhaps this should just be added to the methods as a `@_groupby_supported` decorator?  I realize that it wouldn't reduce the line count much, but I think it's probably worthwhile.

Reply:
Yup, that's exactly what this is doing 🙂 for context on why I made this change, we saw [failures on `ddf.memory_usage`](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci-22-06/job/cudf/job/prb/job/cudf-gpu-test/CUDA=11.0,GPU_LABEL=driver-450,LINUX_VER=centos7,PYTHON=3.8/157/) because upstream Dask was attempting to call dask-cudf's implementation of `groupby.sum()` on a Series, which isn't supported.

I'm definitely okay with adding a decorator for this, as it should make it easier adding new aggregation methods moving forward - will try that out now